File: MDEV1004-M2023-assignment3B/Server/Models/movie.ts
/**
 * File: movie.ts
 * 
 * Student's Name: Divya Patel 
 * Student ID: 200538095
 * 
 * Date: 23/07/2023
 */



import { Schema, model } from 'mongoose';

// Defines the Mongoose schema and model for the Movie collection

interface IMovie {
    movieID: string,
    title: string,
    studio: string,
    genres: string[],
    directors: string[],
    writers: string[],
    actors: string[],
    year: number,
    length: number,
    shortDescription: string,
    mpaRating: string,
    posterLink: string,
    criticsRating: number
}



let movieSchema = new Schema<IMovie>({
    movieID: String,
    title: String,
    studio: String,
    genres: [String],
    directors: [String],
    writers: [String],
    actors: [String],
    year: Number,
    length: Number,
    shortDescription: String,
    mpaRating: String,
    posterLink: String,
    criticsRating: Number
});


// Creating a Movie model using the movie schema

let Movie = model<IMovie>('Movie', movieSchema);

export default Movie;

File: MDEV1004-M2023-assignment3/MDEV1004-M2023-assignment3/Movie.swift
//
//  Movie.swift
//  MDEV1004-M2023-assignment3
//
//  Created by Divya Patel on 2023-07-23.
//

import Foundation

struct Movie: Codable {
    let movieID: String?
    let title: String?
    let studio: String?
    let genres: [String]?
    let directors: [String]?
    let writers: [String]?
    let actors: [String]?
    let year: Int?
    let length: Int?
    let shortDescription: String?
    let mpaRating: String?
    let criticsRating: Double?
}
